### ISSUE TITLE
How do I clear the cache with explicit tokens

### ISSUE THREAD

**Issue by GREATwo** [cameron314/concurrentqueue#287]
1. If multiple threads are enqueued without explicit tokens, how to ensure that the order of enqueueing is the same as the order of enqueueing

2. How can I clear the cache if I use explicit tokens

3. In the latest version, implicit tokens are used to free up memory, which is nice

**Comment by cameron314** [comment]
1. With or without explicit tokens, the final dequeue order is not guaranteed. Only the ordering between elements enqueued by the same thread (implicit) or token (explicit) is preserved. This is covered in the README.
2. If you're asking how to clear the queue in a thread-safe way, you must dequeue all elements. If you're referring to reducing memory usage after elements have been dequeued: destroying the token will free *not* free any excess blocks (which account for most of the memory usage). They will be recycled when another token is created.
3. Yes :-)

**Comment by GREATwo** [comment]
If multiple threads share a producer token and lock the token, you can ensure that enqueueing and enqueueing are consistent

**Comment by eata7** [comment]
> If multiple threads share a producer token and lock the token, you can ensure that enqueueing and enqueueing are consistent

Then what's the point of using this data structure? Use std:queue and lock the whole thing instead.

**Comment by cameron314** [comment]
Yes, at that point locking during enqueue with `ReaderWriterQueue` may be better, if you truly need inter-producer order stability.
